tests/raftstore/cluster.rs

```diff
@@ -297,7 +298,8 @@ impl<T: Simulator> Cluster<T> {
 
         let err = resp.get_header().get_error();
         if !err.has_not_leader() {
-            return false;
+            self.reset_leader_of_region(region_id);
```

Comment:
why not leader doesn't reset but others do? 
If we get stale epoch error, the peer is still leader, why reset?